src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java

```diff
@@ -351,5 +352,76 @@ static UMessageType getMessageType(String ce_type){
                 return UMessageType.UMESSAGE_TYPE_UNSPECIFIED;
         }
     }
+    /**
+     * Get the UMessage from the cloud event
+     * @param event The CloudEvent containing the data.
+     * @return returns the UMessage
+     */
+    static UMessage toMessage(CloudEvent event) {
+        UUri source = LongUriSerializer.instance().deserialize(getSource(event));
+        UPayload payload = UPayload.newBuilder().setFormat(UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF)
+                .setValue(getPayload(event).toByteString()).build();
+        UAttributes.Builder builder =
+                UAttributes.newBuilder().setId(LongUuidSerializer.instance().deserialize(event.getId()))
+                        .setType(getMessageType(event.getType())).setTtl(getCommunicationStatus(event))
```

Comment:
ttl is not commstatus, they are two different attributes.

Reply:
Done